### PR TITLE
12 Elements lists filters and nested elements

### PR DESCRIPTION
**Pull request by JorgeVQA**
Ready point #12.... Please take a look and give me your thoughts.

Thanks in advance

PS/ Changed extension to index from .js to .ts to avoid eslint warnings, and last minute fixes.

### CODE REVIEW

File: cypress/page/products-list.page.ts [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
   
   constructor() {
     this.productToAdd = ".ajax_add_to_cart_button > span";
     this.proceedToCheckoutBtnPopUp = ".button-container > .button-medium > span";
+    this.elementHTML = ".product-container";
   }
   
-  public addProductToCart(): void {
-    cy.get(this.productToAdd).click()
+  findProductByName(prodName: string) {
```
Comment (manuelq12) [inline cypress/page/products-list.page.ts]: I believe this is a private method

File: cypress/page/products-list.page.ts [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
+    return cy.get(this.elementHTML).filter(`:contains(${prodName})`);
```
Comment (manuelq12) [inline cypress/page/products-list.page.ts]: element HTML is not a proper name for the variable, you should change it to something more descriptive

**Review by manuelq12 — CHANGES_REQUESTED** [review]

File: cypress/page/products-list.page.ts (hunk 1 above)
Comment (JorgeVQA) [reply cypress/page/products-list.page.ts]: Done... thanks for the advise.

File: cypress/page/products-list.page.ts (hunk 2 above)
Comment (JorgeVQA) [reply cypress/page/products-list.page.ts]: Done... changed for a better named one.

**Comment by JorgeVQA** [comment]
Fixes were done as requested.   There are other fixes in some "Quotes" I was trying to use... some of them weren't able to use, so I reversed.

Anyway I ran the code and worked as intended.

Please check when you have some free time.

**Review by manuelq12 — APPROVED** [review]

**Pull request merged**